manim/mobject/types/vectorized_mobject.py

```diff
@@ -482,24 +489,24 @@ def add_smooth_curve_to(self, *points):
             ])
         return self
 
-    def has_new_path_started(self):
+    def has_new_path_started(self) -> bool:
         nppcc = self.n_points_per_cubic_curve  # 4
         return len(self.points) % nppcc == 1
 
-    def get_last_point(self):
+    def get_last_point(self) -> Vector:
         return self.points[-1]
 
-    def is_closed(self):
+    def is_closed(self) -> bool:
         return self.consider_points_equals(
             self.points[0], self.points[-1]
         )
 
-    def add_points_as_corners(self, points):
+    def add_points_as_corners(self, points: Iterable[Vector]) -> Iterable[Vector]:
```

Comment:
This doesn't specify that it's precisely the same iterable type that we specified. With this type definition, we could, for example, be given a zip and return a list. We should be as specific as possible with return types. Therefore, my previous suggestion is kept:
```suggestion
    IV = typing.TypeVar("IV", bound=Iterable[Vector])
    def add_points_as_corners(self, points: IV) -> IV:
```
This specifies that the type returned is the exact same as the type specified, while also restricting it to `Iterable[Vector]`.